### ISSUE TITLE
SigV4, EC2 SD: use shared config file by default

### ISSUE THREAD

**Issue by rfratto** [prometheus/prometheus#8592]
#8509 introduced support for SigV4, where most of the logic configuring the AWS SDK (v1) was copied from the EC2 SD code. A [comment](https://github.com/prometheus/prometheus/pull/8509#discussion_r581366746) from @rakyll noted that by default, the configuration in `~/.aws/config` is not loaded unless `AWS_SDK_LOAD_CONFIG` is set to a non-empty string in the environment variables. 

The dependency of `AWS_SDK_LOAD_CONFIG` changes based on the specific AWS SDK being used. For example, the Go _v1_ and Javascript SDKs need it defined, but the Java and Go _v2_ SDKs don't. I'm not an expert on the SDKs, but I think it's becoming expected user behavior that the shared config files are loaded by default. 

Since SigV4 was written with the intent of sharing config settings with the EC2 SD config, I propose we should update both of them to change the default behavior. This can be done in one of two ways: 

1. When instantiating the AWS config, always enable the shared config by appropriately configuring the [SharedConfigState](https://pkg.go.dev/github.com/aws/aws-sdk-go@v1.34.29/aws/session#SharedConfigState).
2. Upgrade usage of the AWS SDK to v2, as recommended [by this comment in the same thread](https://github.com/prometheus/prometheus/pull/8509#discussion_r581598451).

Moving to the newer version of the SDK makes the most sense to me, since I presume that's where most development effort will eventually end up. 

I'm planning on making this change myself next week, but I wanted to open this as an issue first for discussion in case there are problems with the v2 SDK (and to make sure I don't forget to do the work I promised :)

**Comment by roidelapluie** [comment]
This would be a big change in the way we deal with service discovery credentials/configuration. I think we should wait enough time to let people express their interest or cons.

**Comment by rfratto** [comment]
Good point, I'll hold off for a while. Any concerns about getting this change into the upcoming release? (with the consideration that the workaround is to supply `AWS_SDK_LOAD_CONFIG`)

**Comment by roidelapluie** [comment]
I see two challenges:
-  be careful about the priority of existing env variables regarding this, to not break users
- find a good source of documentation for this

**Comment by matthiasr** [comment]
My understanding of the behavior so far has been that, if no authentication information is given in the configuration file, we defer to the SDK. Since this is the default SDK behavior in the latest version, we should not get in the way of it. Upgrading the SDK, if feasible, deems like the cleanest option – Prometheus continues to not have an opinion on where the SDK gets its configuration, users get exactly the same behavior as from other tools.

We should call this out as a change ("the `AWS_SDK_LOAD_CONFIG` workaround is no longer necessary") but I don't think we need to reproduce documentation of the standard SDK behavior. We should add to the docs that the default SDK behavior applies if no explicit options are given; that is already true today.

**Comment by roidelapluie** [comment]
I am wondering if the file would override env variables or not. I agree we don't need to duplicate the docs, but we need to find and link to accurate docs for ursers.

**Comment by matthiasr** [comment]
Is this the right one?

https://docs.aws.amazon.com/credref/latest/refdocs/overview.html

**Comment by roidelapluie** [comment]
Yesssss thanks

**Comment by mrkwtz** [comment]
As someone who wanted to use the AWS config profile option with sigv4 please consider mentioning the `AWS_SDK_LOAD_CONFIG` env variable in the docs. I just spent a day figuring it out why it just wouldn't use the AWS config file. 

If you're using the remote_write option with sigv4 in an AWS environment it'll always try to use the WebToken or the IAM role authentication (that would be the default credentials provider chain) but given that there is a real possibility that the remote_write URL is outside of your control (e.g. a third party cost allocation tool) means that these authentication options won't work and you need to use explicit credentials (e.g. the AWS config file).

**Comment by gudladona** [comment]
It also appears that, for non-aws environments,  there is no logic for the remote write client to reload the credentials from the config file periodically, It's a common usecase that the credentials are refreshed and re-written to the same file, to be reloaded up by the applications. The expectation seems to be that the credentials are permanent unless explicitly revoked. Did anyone else encounter this?

**Comment by LeviHarrison** [comment]
If you mean when using non-SigV4 authentication methods, various other authentication methods have an option to specify a file for their respective credentials (`basic_auth.password_file`, `authorization.credentials_file`, etc.) And the credentials in those files are checked for changes on every request.

**Comment by LeviHarrison** [comment]
On another note, @roidelapluie I don't think any concerns have been raised that would that would make us reconsider adding this feature, and there certainly is some interest, so maybe we could upgrade it to a p3?

**Comment by beorn7** [comment]
Hello from the bug scrub.

@LeviHarrison @roidelapluie nothing has happened here in a while. What's the state of this? Is it still worth pursuing? Should we bump the priority or tag it with "help needed", or rather close?
